src/Storages/MergeTree/ReplicatedMergeTreeLogEntry.cpp

```diff
@@ -240,7 +240,7 @@ void ReplicatedMergeTreeLogEntryData::readText(ReadBuffer & in)
 
                 if (checkString("merge_type: ", in))
                 {
-                    UInt64 value;
+                    UInt32 value;
```

Comment:
Merge type is very small, and checkAndGetMergeType() (used just below) accept UInt32, the whole MergeType is enum of 3 elements actually, so I wouldn't expect to have something different there, except in case of corruption, in this case checkAndGetMergeType will throw an exception.

Reply:
You are right, I misread the code.